lib/services/security/users.js

```diff
@@ -228,20 +243,27 @@ function __prepareUserForUpsert(user) {
 
 function __upsertUser(user) {
   return new Promise((resolve, reject) => {
+    let permissions = this.utilsService.clone(user.permissions);
     this.__prepareUserForUpsert(user).then(prepared => {
+      delete prepared.permissions; //Don't store in DB
       this.dataService.upsert(
         '/_SYSTEM/_SECURITY/_USER/' + prepared.username,
         prepared,
         { merge: true },
-        (e, result) => {
+        async (e, result) => {
```

Comment:
note that async functions used as callbacks can result in uncaught promise rejections, which we now want to eliminate. You can achieve that by wrapping the entire callback function in a try/catch block.